### PR TITLE
Prepare v3.14 (part 2)

### PR DESCRIPTION
**Pull request by avagin**
* Add support of the cgroup v2 freezer.
* Add memfd support.
*  Clean up compel headers.
* Adds support of opened O_PATH fds.

### CODE REVIEW

File: criu/util.c [hunk 1]
```diff
@@ -30,6 +30,8 @@
 
 #include "linux/mount.h"
 
+#include "linux/mount.h"
+
```
Comment (Snorch) [inline criu/util.c]: Duplicate include statement. 4997a096e ("util: introduce the mount_detached_fs helper") Is already there.

File: criu/cr-restore.c [hunk 2]
```diff
@@ -76,6 +75,8 @@
 #include "sigframe.h"
 #include "fdstore.h"
 #include "string.h"
+#include "memfd.h"
+#include "string.h"
```
Comment (Snorch) [inline criu/cr-restore.c]: Again include statement duplicate. We already have bc49927bb ("criu: Make use strlcpy() to copy into allocated strings"). More over we have ```#include <string.h>``` three times in this c file.

**Comment by avagin** [comment]
> Maybe you can also consider to take some fixes in next versions:
> 36e1da9 ("restorer/inotify: Don't overflow PIE stack")
> f728391 ("restorer/inotify: reorder inotify cleanup after waiting helpers and zombies")
> 48ae837 ("cgroup: fix cg_yard leak on error path in prepare_cgroup_sfd")
> be22689 ("criu: fix leaks detected by coverity scan")

All these patches are in the master branch already.

**Comment by Snorch** [comment]
> > Maybe you can also consider to take some fixes in next versions:
> > [36e1da9](https://github.com/checkpoint-restore/criu/commit/36e1da911ac1e9e80c37e45e9d99d6f4a9311e5b) ("restorer/inotify: Don't overflow PIE stack")
> > [f728391](https://github.com/checkpoint-restore/criu/commit/f728391d3633d1133ef3cd2ea5a20aaaade71ebe) ("restorer/inotify: reorder inotify cleanup after waiting helpers and zombies")
> > [48ae837](https://github.com/checkpoint-restore/criu/commit/48ae837908283f444f0ab29ae8800f03eba6223f) ("cgroup: fix cg_yard leak on error path in prepare_cgroup_sfd")
> > [be22689](https://github.com/checkpoint-restore/criu/commit/be22689de25c897461ebaa166a05fb7070104845) ("criu: fix leaks detected by coverity scan")
> 
> All these patches are in the master branch already.

Thanks for pointing this out, somehow missed this.

**Pull request merged**